### PR TITLE
Fix #1590: Update licenses file. Add licenses setting option.

### LINKED ISSUE #1590: Add Licenses setting option.
We should allow to present open source licenses we use to users.

### PR DESCRIPTION
**Pull request by iccub**
## Summary of Changes

This pull request fixes issue #1590 

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`

## Test Plan:

Open Settings -> About -> Licenses. Confirm it contains MPL license and licenses of dependencies

## Screenshots:

Zrzut ekranu 2019-10-23 o 20 40 04

Zrzut ekranu 2019-10-23 o 20 40 16

## Reviewer Checklist:

- [x] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `release-notes/(include|exclude)`
  - `bug` / `enhancement`
- [x] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [x] Adequate unit test coverage exists to prevent regressions.
- [x] Adequate test plan exists for QA to validate (if applicable).
- [x] Issue is assigned to a milestone (should happen at merge time).

### CODE REVIEW

File: Client/Assets/About/Licenses.html [hunk 1]
```diff
@@ -249,21 +249,17 @@ <h4 id="exhibit-b---incompatible-with-secondary-licenses-notice">Exhibit B - “
...
+<!-- Carthage -->
 
 <center><p>-</p></center>
 
-<!-- -->
 <div>
-<input id="ac-11" name="accordion-1" type="checkbox" />
-<label for="ac-11"><h2>Alamofire</h2></label>
+<input id="ac-Deferred" name="accordion-1" type="checkbox" />
+<label for="ac-Deferred"><h2>Deferred</h2></label>
```
Comment (iccub) [inline Client/Assets/About/Licenses.html:258]: I ordered dependency licenses to match structure of cartfile

**Review by Brandon-T — APPROVED** [review]
Hmm it flickers a little when I expand the cells and they have a weird padding but I think this is just due to HTML being rendered instead of native.

Also, if the cell is at the bottom of the screen and I expand, it doesn't push up atm. I guess we can fix this later. For now, code is good.

**Comment by iccub** [comment]
Blocking until @jhreis review and/or sec review

**Comment by iccub** [comment]
@Brandon-T yeah, I didn't change the html implementation from what we had from upstream

**Comment by jhreis** [comment]
This looks programmatically good to me. I am waiting on some final wording, so not merging right now.

update: received confirmation that this is gtg for now (see https://github.com/brave/brave-ios/pull/1780#issuecomment-545711662 for more info)

**Review by jhreis — APPROVED** [review]

**Comment by jhreis** [comment]
We very likely will improve upon this in the future (e.g. better text information, tag linking), but this meets what we need so merging this.

**Pull request merged**